dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/CompanyDataControllerGetCompaniesEndpointTest.kt

```diff
@@ -0,0 +1,287 @@
+        )
+        val companyInformation = CompanyInformation(
+            "retrieve empty search string", "", companyIdentifier, "",
+            listOf(),
+        )
+        val uploadedCompany = apiAccessor.companyDataControllerApi.postCompany(companyInformation)
+        val uploadedData = uploadTestDataSet(uploadedCompany.companyId).copy(uploaderUserId = null)
+        val expectedCompany = StoredCompany(
+            uploadedCompany.companyId,
+            uploadedCompany.companyInformation,
+            listOf(uploadedData),
+        )
+
+        val retrievedCompanies = apiAccessor.getCompaniesOnlyByName("")
+        assertTrue(
+            retrievedCompanies.contains(expectedCompany),
+            "Not all the companyInformation of the posted companies could be found in the stored companies.",
+        )
+    }
+
+    @Test
+    fun `search for identifier and name substring to verify substring matching in company search`() {
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Uploader)
+        val testIdentifier = UUID.randomUUID().toString()
+        val testName = "SubstringSearch"
+        val companyIdentifier = listOf(
+            apiAccessor.createCompanyIdentifier(CompanyIdentifier.IdentifierType.lei, testIdentifier),
+        )
+        val companyInformation = CompanyInformation(
+            testName, "", companyIdentifier, "", listOf(),
+        )
+        val uploadedCompany = apiAccessor.companyDataControllerApi.postCompany(companyInformation)
+        val uploadedData = uploadTestDataSet(uploadedCompany.companyId).copy(uploaderUserId = null)
+        val expectedCompany = StoredCompany(
+            uploadedCompany.companyId,
+            uploadedCompany.companyInformation,
+            listOf(uploadedData),
+        )
+
+        val searchIdentifier = testIdentifier.drop(1).dropLast(1)
+        val searchName = testName.drop(1).dropLast(1)
+        val searchResponseForName = apiAccessor.getCompaniesOnlyByName(searchName)
+        val searchResponseForIdentifier = apiAccessor.getCompaniesByNameAndIdentifier(searchIdentifier)
+        assertTrue(
+            searchResponseForName.contains(expectedCompany),
+            "The search results do not contain the expected company.",
+        )
+        assertTrue(
+            searchResponseForIdentifier.contains(expectedCompany),
+            "The search results do not contain the expected company.",
+        )
+    }
+
+    @Test
+    fun `search for name and check the ordering of results`() {
+        val testString = "unique-test-string-${UUID.randomUUID()}"
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Uploader)
+        val companyList = createCompaniesForTestingOrdering(testString)
+        for (company in companyList) {
+            val uploadedCompany = apiAccessor.companyDataControllerApi.postCompany(company)
+            uploadTestDataSet(uploadedCompany.companyId)
+        }
```

Comment:
Man könnte den ganzen Block hier direkt in der Methode machen und sich hier Code sparen.  Wäre etwas übersichtlich vllt

Reply:
Dann wird die Methode vermutlich wieder länger als unser LoC Limit.